app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.java

```diff
@@ -371,6 +373,18 @@ void validateText() {
         updateSaveButtonEnabled();
     }
 
+    private void checkKeyboardLanguage(String text) {
+        FirebaseLanguageIdentification languageIdentifier =
```

Comment:
Ideally anything that references Firebase or GMS should be in a separate file, so that it can be stubbed out for builds that won't use Google Play Services.